src/enclave.rs

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Enclave object
+
+extern crate std;
+
+use crate::page::{Class, Flags, SecInfo, Secs};
+use crate::signature::Signature;
+
+use std::fs::{File, OpenOptions};
+use std::io::Result;
+use std::marker::PhantomData;
+
+use iocuddle::{Group, Ioctl, Write, WriteRead};
+
+/// Wraps an enclave file descriptor.
+pub struct Enclave(File);
+
+impl From<Enclave> for File {
+    fn from(enclave: Enclave) -> File {
+        enclave.0
+    }
+}
+
+impl Enclave {
+    /// The path of the device file.
+    pub const DEVICE: &'static str = "/dev/sgx_enclave";
+
+    const SGX: Group = Group::new(0xA4);
+
+    /// Create a new instance.
+    pub fn new() -> Result<Self> {
+        OpenOptions::new()
+            .read(true)
+            .write(true)
+            .open(Enclave::DEVICE)
+            .map(Self)
+    }
+
+    pub fn try_clone(&mut self) -> Result<Self> {
+        Ok(Self(self.0.try_clone()?))
+    }
+
+    /// ENCLS[ECREATE] wrapper
+    pub fn create(&mut self, secs: &Secs) -> Result<()> {
+        const CREATE: Ioctl<Write, &'static Create<'static>> = unsafe { Enclave::SGX.write(0x00) };
+        CREATE
+            .ioctl(&mut self.0, &Create(secs as *const _ as _, PhantomData))
+            .map(|_| ())
```

Comment:
For what reason are we deliberately throwing away the success value in all of these methods?

Reply:
They just return zero on success.